qiita_pet/templates/show_analyses.html

```diff
@@ -51,6 +55,16 @@
       {% end %}
         </td>
         <td>
+          {% if mappings[_id] is not None %}
+            {% raw download_link_or_path(is_local_request, mappings[_id]['mapping'], mappings[_id]['id'], 'mapping file') %}
```

Comment:
We have tried to remove as much python code from the templates as possible. Can you move this to the handler itself? At the end, what the template wants is for each analysis have a tuple with 2 elements, the mapping file (either link or filepath) and the list of biom tables...